ops/manifests/manifest.py

```diff
@@ -188,21 +188,27 @@ def resources(self) -> KeysView[HashableResource]:
             (HashableResource(obj), None) for obj in all_resources
         ).keys()
 
+    def _resource_from_yaml(self, filepath: Path) -> List[AnyResource]:
+        def create_crd(rsc):
+            if rsc.kind == "CustomResourceDefinition":
+                create_resources_from_crd(rsc)
+            return rsc
+
+        return [
+            create_crd(codecs.from_dict(item))  # Map to lightkube resources
+            for item in self._safe_load(filepath)
+        ]
+
     @lru_cache()
-    def _safe_load(self, filepath: Path) -> List[AnyResource]:
+    def _safe_load(self, filepath: Path) -> List[Mapping]:
         """Read manifest file and parse its content into lightkube objects.
 
         Lightkube can't properly read manifest files which contain List kinds.
         """
         content = filepath.read_text()
 
-        def create_crd(rsc):
-            if rsc.kind == "CustomResourceDefinition":
-                create_resources_from_crd(rsc)
-            return rsc
-
         return [
-            create_crd(codecs.from_dict(item))  # Map to lightkube resources
+            rsc
```

Comment:
this is the guts of the change

the `@lru_cache()` was doing the right thing -- caching the read lightkube objects that were loaded from the filesystem.  Both if those had been patched once, they'd be patched AGAIN on the second call to the `.resources` property yielding odd results. 

this splits the reading from the filesystem from the making LK objects from dicts.  This way the patches are only done on ephemeral objects.